undefined: math.MaxInt with Go 1.16

Recent changes have introduced the use of stdlib's math.MaxInt, see e.g. here:
https://github.com/ericlagergren/decimal/pull/167/files#diff-1ebe9d76178a47d079967d181e236d63579a3be7aa6596f39b1a013badbe1979R772

The new math constants have only been introduced in 1.17 and as of 1.16.9 aren't available in 1.16:
https://go-review.googlesource.com/c/go/+/247058/

Your go.mod states 1.16, leading to undescriptive errors when using decimal from 1.16. If I understood it correctly, increasing the go directive would lead to a reasonable error message, but would break your promise of supporting the the most recent major Go versions?

@ericlagergren Thanks for your quick response!
Confirmed working with `go1.16.9`
```
❯ go version
go version go1.16.9 darwin/amd64
```

```
❯ go get github.com/ericlagergren/decimal
go: downloading github.com/ericlagergren/decimal v0.0.0-20211022110326-b30d25f2d547
# github.com/ericlagergren/decimal
../../go/pkg/mod/github.com/ericlagergren/decimal@v0.0.0-20211022110326-b30d25f2d547/big_ctx.go:772:27: undefined: math.MaxInt
../../go/pkg/mod/github.com/ericlagergren/decimal@v0.0.0-20211022110326-b30d25f2d547/big_ctx.go:3179:27: undefined: math.MaxInt
../../go/pkg/mod/github.com/ericlagergren/decimal@v0.0.0-20211022110326-b30d25f2d547/scan.go:312:16: undefined: math.MaxInt
```

```
❯ go get github.com/ericlagergren/decimal@b59ce5967ce2721644e1a771cd70d8d4b71a484c
go: downloading github.com/ericlagergren/decimal v0.0.0-20211026174116-b59ce5967ce2
go get: added github.com/ericlagergren/decimal v0.0.0-20211026174116-b59ce5967ce2
```

--- a/big_ctx.go
+++ b/big_ctx.go
@@ -769,7 +769,7 @@ func (c Context) prepCosine(z, x *Big) (*Big, int, bool) {
 		// 1+((v-1)/10) will be wildly incorrect for v == 0, but x/y = 0 iff
 		// x = 0 and y != 0. In this case, -2pi <= x >= 2pi, so we're fine.
 		prec, carry := bits.Add64(1+((uv-1)/10), uint64(c.Precision), 0)
-		if carry != 0 || prec > math.MaxInt {
+		if carry != 0 || prec > arith.MaxInt {
 			return nil, 0, false
 		}
 		pctx := Context{Precision: int(prec)}
@@ -3176,7 +3176,7 @@ func (c Context) prepTan(z, x *Big) (*Big, bool) {
 		// 1+((v-1)/10) will be widly incorrect for v == 0, but x/y = 0 iff
 		// x = 0 and y != 0. In this case, -2pi <= x >= 2pi, so we're fine.
 		prec, carry := bits.Add64(1+((uv-1)/10), uint64(c.Precision), 0)
-		if carry != 0 || prec > math.MaxInt {
+		if carry != 0 || prec > arith.MaxInt {
 			return nil, false
 		}
 		pctx := Context{Precision: int(prec)}

--- a/dectest/test.go
+++ b/dectest/test.go
@@ -3,13 +3,13 @@ package dectest
 import (
 	"fmt"
 	"io"
-	"math"
 	"os"
 	"reflect"
 	"strconv"
 	"testing"
 
 	. "github.com/ericlagergren/decimal"
+	"github.com/ericlagergren/decimal/internal/arith"
 )
 
 func Test(t *testing.T, file string) {
@@ -182,7 +182,7 @@ func execute(t *testing.T, c *Case) {
 			check(t, z.Set(Min(x, y)), r, c, flags)
 		case OpQuantize:
 			v, _ := y.Int64()
-			if v > math.MaxInt {
+			if v > arith.MaxInt {
 				t.Logf("%s: int out of range: %d", c.ID, v)
 				return
 			}

--- a/internal/arith/arith.go
+++ b/internal/arith/arith.go
@@ -7,6 +7,11 @@ import (
 	"math/bits"
 )
 
+const (
+	intSize = 32 << (^uint(0) >> 63) // 32 or 64
+	MaxInt  = 1<<(intSize-1) - 1
+)
+
 // Words returns a little-endian slice of big.Words representing
 // the uint64.
 func Words(x uint64) []big.Word {

--- a/scan.go
+++ b/scan.go
@@ -309,7 +309,7 @@ func (z *Big) scanExponent(r io.ByteScanner) error {
 		r.UnreadByte()
 	}
 
-	max := uint64(math.MaxInt)
+	max := uint64(arith.MaxInt)
 	if neg {
 		max++ // -math.MinInt
 	}

